### PR TITLE
Set GMT_COMPATIBILITY to 6 when pygmt session starts

### LINKED ISSUE #428: pygmt-session [ERROR]: GMT_COMPATIBILITY: Expects values from 6 to 6; reset to 6.
pygmt continues to provide annoying errors about GMT_COMPATIBILITY. These can arise in many circumstances (such as in jupyter notebooks), but here is the easiest one:

```
In [1]: import pygmt
In [2]: exit
pygmt-session [ERROR]: GMT_COMPATIBILITY: Expects values from 6 to 6; reset to 6.
>  gmt --version
6.0.0
> brew info gmt
gmt: stable 6.0.0 (bottled), HEAD
Tools for manipulating and plotting geographic and Cartesian data
https://www.generic-mapping-tools.org/
/usr/local/Cellar/gmt/6.0.0_5 (1,584 files, 150.6MB) *
```
OS: macos
pygmt: v0.1 (installed via pypi)
gmt: installed by brew (v. 6.0.0_5)
python: 3.7.7

### PR DESCRIPTION
**Pull request by weiji14**
**Description of proposed changes**

To prevent users getting "pygmt-session [ERROR]: GMT_COMPATIBILITY: Expects values from 6 to 6; reset to 6". Overrides [GMT_COMPATIBILITY](https://docs.generic-mapping-tools.org/6.0/gmt.conf.html#gmt-miscellaneous-parameters) setting in users `gmt.conf` file.

Fixes #428, #365

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [x] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

File: pygmt/session_management.py [hunk 1]
```diff
@@ -15,6 +15,7 @@ def begin():
     prefix = "pygmt-session"
     with Session() as lib:
         lib.call_module("begin", prefix)
+        lib.call_module("set", "GMT_COMPATIBILITY 6")
 
 
 def end():
```
Comment (weiji14) [inline pygmt/session_management.py:21]: Can't find a good way to reset the `GMT_COMPATIBILITY` setting back to what's in the original `gmt.conf`. Probably doesn't matter too much?

File: pygmt/tests/test_session_management.py [hunk 2]
```diff
@@ -20,3 +20,35 @@ def test_begin_end():
...
+    try:
+        with Session() as lib:
+            # pretend that gmt.conf has GMT_COMPATIBILITY = 5
+            lib.call_module("gmtset", "GMT_COMPATIBILITY 5")
+        begin()
+        with Session() as lib:
+            lib.call_module("basemap", "-R10/70/-3/8 -JX4i/3i -Ba")
+            out, err = capsys.readouterr()  # capture stdout and stderr
+            assert out == ""
+            assert err != (
```
Comment (weiji14) [inline pygmt/tests/test_session_management.py:40]: Note: To future maintainers on what `capsys` is. It's a pytest fixture for capturing stdout/stderr. See https://docs.pytest.org/en/5.4.2/capture.html#accessing-captured-output-from-a-test-function

File: pygmt/session_management.py (hunk 1 above)
Comment (seisman) [reply pygmt/session_management.py:21]: See https://github.com/GenericMappingTools/pygmt/issues/428#issuecomment-629809995

File: pygmt/session_management.py [hunk 3]
```diff
@@ -15,6 +15,7 @@ def begin():
     prefix = "pygmt-session"
     with Session() as lib:
         lib.call_module("begin", prefix)
+        lib.call_module("set", "GMT_COMPATIBILITY 6")
```
Comment (seisman) [inline pygmt/session_management.py:18]: Perhaps add a comment here why pygmt needs GMT_COMPATIBILITY=6.

Comment (seisman) [inline pygmt/session_management.py:18]:
```suggestion
        lib.call_module("begin", prefix)
        # pygmt replies on GMT modern mode, which requires GMT_COMPATIBILITY to 6
        lib.call_module("set", "GMT_COMPATIBILITY 6")
```

File: pygmt/tests/test_session_management.py [hunk 4]
```diff
@@ -20,3 +20,35 @@ def test_begin_end():
...
+            )
+            assert err == ""  # double check that there are no other errors
+    finally:
+        with Session() as lib:
+            # revert gmt.conf back to GMT_COMPATIBILITY = 6
+            lib.call_module("set", "GMT_COMPATIBILITY 6")
+        end()
+        begin()  # Restart the global session
+        assert os.path.exists("gmt.conf")
+        os.remove("gmt.conf")
```
Comment (seisman) [inline pygmt/tests/test_session_management.py:52]: I don't think there is a "gmt.conf" file in the current directory. It should be in the temporary gmt session directory.

Comment (weiji14) [reply pygmt/tests/test_session_management.py:52]: I kinda agree, but wouldn't the assertion fail if there isn't a `gmt.conf` file? :confused: Things got really confusing for me when I was designing this test with `gmt.conf` files all over the place, one kept popping up in my `pygmt` folder with `GMT_COMPATIBILITY 5` for some reason.

**Comment by weiji14** [comment]
@MarkWieczorek, do you want to test this branch (using `pip install https://github.com/GenericMappingTools/pygmt/archive/gmt_compat_6.zip`)? Just want to make sure that it fixes your issue.

**Comment by MarkWieczorek** [comment]
It works!

File: pygmt/tests/test_session_management.py (hunk 4 above)
Comment (weiji14) [reply pygmt/tests/test_session_management.py:52]: Ok, you're right, it's in "/home/.../pygmt/tmp-test-dir-with-unique-name/gmt.conf". I'll remove these lines once the tests on Windows pass in the other PR.

Comment (seisman) [reply pygmt/tests/test_session_management.py:52]: I still don't understand why the test works, but since the test pass, OK to merge.

Comment (weiji14) [reply pygmt/tests/test_session_management.py:52]: Yeah, it's a complicated test. but removing the line at https://github.com/GenericMappingTools/pygmt/blob/8df13da7d7163ed265c22b7de18e1df344ec3664/pygmt/session_management.py#L19 should fail this test, as encountered by Mark.

**Pull request merged**